stetho/src/main/java/com/facebook/stetho/inspector/protocol/ScreencastDispatcher.java

```diff
@@ -0,0 +1,142 @@
+  private boolean mIsRunning;
+  private static final float FRAMERATE = 60f;
+  private static final long FRAME_DELAY = (long) (1000 / FRAMERATE);
+
+  private Handler mBackgroundHandler;
+  private JsonRpcPeer mPeer;
+
+  public ScreencastDispatcher() {
+    mHandlerThread.start();
+  }
+
+  public void startScreencast(JsonRpcPeer peer) {
+    LogUtil.d("Starting screencast");
+    mPeer = peer;
+    mIsRunning = true;
+    mBackgroundHandler = new Handler(mHandlerThread.getLooper());
+    mMainHandler.postDelayed(mBitmapFetchRunnable, FRAME_DELAY);
+  }
+
+  public void stopScreencast() {
+    LogUtil.d("Stopping screencast");
+    mMainHandler.removeCallbacks(mBitmapFetchRunnable);
+    mBackgroundHandler.removeCallbacks(mEventDispatchRunnable);
+    mIsRunning = false;
+  }
+
+  private class BitmapFetchRunnable implements Runnable {
+    @Override
+    public void run() {
+      mBackgroundHandler.post(mEventDispatchRunnable
+          .withBitmap(getScreenBitmap())
+          .withEndAction(this));
+    }
+
+    private Bitmap getScreenBitmap() {
+      if (!mIsRunning) {
+        return null;
+      }
+      Activity activity = mActivityTracker.tryGetTopActivity();
+      if (activity == null) {
+        return null;
+      }
+      // This stuff needs to happen in the UI thread
+      View rootView = activity.getWindow().getDecorView();
+      rootView.setDrawingCacheEnabled(true);
+      Bitmap drawingCache = rootView.getDrawingCache();
+      Bitmap bitmap = android.graphics.Bitmap.createBitmap(drawingCache);
+      rootView.setDrawingCacheEnabled(false);
+      return bitmap;
+    }
+  }
+
+  private class EventDispatchRunnable implements Runnable {
+    private Bitmap mBitmap;
+    private Runnable mEndAction;
+
+    private EventDispatchRunnable withBitmap(Bitmap bitmap) {
+      mBitmap = bitmap;
+      return this;
+    }
+
+    private EventDispatchRunnable withEndAction(Runnable endAction) {
+      mEndAction = endAction;
+      return this;
+    }
+
+    @Override
+    public void run() {
+      if (!mIsRunning) {
+        return;
+      }
+      ByteArrayOutputStream stream = new ByteArrayOutputStream();
+      mBitmap.compress(Bitmap.CompressFormat.JPEG, 80, stream);
+      mEvent.data = Base64.encodeToString(stream.toByteArray(), Base64.DEFAULT);
+      mMetadata.offsetTop = 0;
+      mMetadata.scrollOffsetX = 0;
+      mMetadata.scrollOffsetY = 0;
+      mMetadata.pageScaleFactor = 1;
+      mMetadata.deviceWidth = mBitmap.getWidth();
+      mMetadata.deviceHeight = mBitmap.getHeight();
+      mEvent.metadata = mMetadata;
+      mPeer.invokeMethod("Page.screencastFrame", mEvent, null);
+      mMainHandler.postDelayed(mEndAction, FRAME_DELAY);
+    }
+  }
+
+  private static class ScreencastFrameEvent {
+    @JsonProperty(required = true)
+    public String data;
+
+    @JsonProperty(required = true)
+    public Metadata metadata;
+
+    private static class Metadata {
```

Comment:
Prefer not to use inner classes for this type of thing.  Also, this should be defined in the `Page` module and made public.